numpy/core/src/multiarray/datetime_strings.c

```diff
@@ -1592,3 +1593,344 @@ array_datetime_as_string(PyObject *NPY_UNUSED(self), PyObject *args,
 
     return NULL;
 }
+
+/*
+ * This is the Python-exposed datetime_strftime function.
+ */
+NPY_NO_EXPORT PyObject *
+array_datetime_strftime(PyObject *NPY_UNUSED(self), PyObject *args,
+                                PyObject *kwds)
+{
+    PyObject *arr_in = NULL, *fmt_in = NULL, *timezone_obj = NULL;
+
+    int local = 0;
+    int utc = 0;
+    PyArray_DatetimeMetaData *meta;
+    int strsize;
+
+    PyArrayObject *ret = NULL;
+
+    NpyIter *iter = NULL;
+    PyArrayObject *op[2] = {NULL, NULL};
+    PyArray_Descr *op_dtypes[2] = {NULL, NULL};
+    npy_uint32 flags, op_flags[2];
+    char *fmt, *temp;
+    struct tm timest;
+
+    static char *kwlist[] = {"arr", "fmt", "timezone", "booger", NULL};
```

Comment:
Leftover booger kwarg?

Reply:
Actually, I couldn't get the kwarg parsing to work with a shorter list.  I'm very much *not* up on the python C API, so cleanup help very much welcome...

Reply:
Ah, OK. I fixed it...